crypto/ec/ecdsa_sign.c

```diff
@@ -29,7 +35,13 @@ ECDSA_SIG *ECDSA_do_sign_ex(const unsigned char *dgst, int dlen,
 int ECDSA_sign(int type, const unsigned char *dgst, int dlen, unsigned char
                *sig, unsigned int *siglen, EC_KEY *eckey)
 {
-    return ECDSA_sign_ex(type, dgst, dlen, sig, siglen, NULL, NULL, eckey);
+#ifndef OPENSSL_NO_CNSM
+
+    if (EC_GROUP_get_curve_name(EC_KEY_get0_group(eckey)) == NID_sm2)
+        return  sm2_sign(dgst, dlen, sig, siglen, eckey);
+     else
```

Comment:
The `else` isn't required because of the return on the previous line (which has a double space after it).